### PR TITLE
[FEAT]: Waitlist API - ADD USER TO WAITLIST

### PR DESCRIPTION
**Pull request by Arnthorny**
## Description
This pull request introduces an endpoint for admins to manually add a user to the waitlist. 

The changes include:
 
* **Endpoint Addition:** Added `POST /api/v1/waitlist/admin` for deactivating invite links.
* **Error Handling:** Implemented checks for invalid data and unauthorized access.
* **Testing:** Added pytest cases to ensure functionality and edge cases are covered.
* **Bug Fixing:** Fixed a bug that affected the login functionality of users.
 
 
## Related Issue (Link to issue ticket)
[Issue Ticket](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/238)
 
 ## Motivation and Context
This feature is essential to allow admins add users to a company's waitlist.
 
 ## How Has This Been Tested?
 * **Unit Tests:** Added tests to validate the addition of users to the database, handling of duplicate requests, and authorization failures.
* **Test Environment:** Tests were run using a mocked database instance to avoid interference with the production environment.

Tests include:
 
* Adding a user to the waitlist successfully.
* Handling attempts to add single user multiple times.
* Handling unauthorized access.

### Successful Request
Screenshot (442)

### Duplicated Request
Screenshot (443)

## Types of changes
* [x]  Bug fix (non-breaking change which fixes an issue)
* [x]  New feature (non-breaking change which adds functionality)
* [ ]  Breaking change (fix or feature that would cause existing functionality to change)
 
 ## Checklist:
 * [x]  My code follows the code style of this project.
 * [x]  My change requires a change to the documentation.
 * [x]  I have updated the documentation accordingly.
 * [x]  I have read the **CONTRIBUTING** document.
 * [x]  I have added tests to cover my changes.
 * [x]  All new and existing tests passed.

### CODE REVIEW

File: api/utils/config.py
Comment (bolexs) [inline api/utils/config.py:1]: is this file not in use?

File: api/utils/dependencies.py [hunk 1]
```diff
@@ -7,11 +7,11 @@
 from datetime import datetime, timedelta
 from api.v1.models.user import User
 import os
-from jose import JWTError
+from jose import JWTError, ExpiredSignatureError
 import bcrypt
 from api.v1.schemas.token import TokenData
 from api.db.database import get_db
-from .config import SECRET_KEY, ALGORITHM
+from api.utils.settings import settings
```
Comment (bolexs) [inline api/utils/dependencies.py:14]: ✅

File: api/utils/dependencies.py [hunk 2]
```diff
@@ -22,14 +22,14 @@ def get_current_user(db: Session = Depends(get_db), token: str = Depends(oauth2_
...
+        user_id: str = payload.get("user_id")
+        print(user_id)
+        if user_id is None:
             raise credentials_exception
-        token_data = TokenData(username=username)
-    except JWTError:
+    except (JWTError, jwt.exceptions.ExpiredSignatureError, jwt.exceptions.DecodeError):
         raise credentials_exception
-    user = db.query(User).filter(User.username == token_data.username).first()
+    user = db.query(User).filter(User.id == user_id).first()
```
Comment (bolexs) [inline api/utils/dependencies.py:32]: is this approach approved by others?

File: api/utils/exceptions.py
Comment (bolexs) [inline api/utils/exceptions.py:1]: please do not comment out
Whatever is not being used should be deleted

File: api/utils/config.py
Comment (Arnthorny) [reply api/utils/config.py:1]: > is this file not in use?

It's not in use as all its functionalities have been moved to the api/utils/settings file.

File: api/utils/dependencies.py (hunk 2 above)
Comment (Arnthorny) [reply api/utils/dependencies.py:32]: The access token created in the routes/auth.py file had always been encoded using the `user_id`. 

The `create_access_token` method of the `UserService`, class takes a `user_id` string as one of its arguments.

File: api/v1/routes/waitlist.py
Comment (bolexs) [inline api/v1/routes/waitlist.py:1]: nice approach ✅

File: api/utils/exceptions.py
Comment (Arnthorny) [reply api/utils/exceptions.py:1]: I will make due changes now.

File: seed2.py
Comment (bolexs) [inline seed2.py:1]: why does the organization not have ids?

File: api/utils/config.py
Comment (bolexs) [reply api/utils/config.py:1]: alright

**Review by bolexs — CHANGES_REQUESTED** [review]
clean up the code base

File: seed2.py
Comment (Arnthorny) [reply seed2.py:1]: > why does the organization not have ids?

All models inherit from the `BaseTableModel` which inherits from the `declarative_base` instance `Base`. 

This `BaseTableModel` implements the id Column, making it unnecessary for any model to be explicitly instantiated with an id argument.

File: api/v1/routes/waitlist.py
Comment (Arnthorny) [reply api/v1/routes/waitlist.py:1]: Appreciate chief

Comment (Arnthorny) [reply api/v1/routes/waitlist.py:1]: > nice approach ✅

Appreciate chief.

**Comment by Arnthorny** [comment]
> clean up the code base

I have removed unnecessary imports and commented out codes from the codebase

**Review by bolexs — APPROVED** [review]

**Review by zxenonx — APPROVED** [review]
Thanks @Arnthorny

**Comment by Arnthorny** [comment]
> Thanks @Arnthorny

Appreciate chief

**Pull request merged**